app/services/worker_monitor.py

```
@@ -0,0 +1,672 @@
+    global _monitor_instance
+    if _monitor_instance is None:
+        _monitor_instance = WorkerMonitor()
+        try:
+            asyncio.run(_monitor_instance.initialize())
+        except RuntimeError:
+            # Event loop already running — reset instance and raise
+            _monitor_instance = None
+            raise RuntimeError(
+                "Cannot initialize worker monitor synchronously in async context. "
+                "Use get_worker_monitor() from async code instead."
+            )
```

Ruff (0.14.14)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)